aviatrix/resource_aviatrix_aws_tgw.go

```diff
@@ -138,16 +156,45 @@ func resourceAviatrixAWSTgwCreate(d *schema.ResourceData, meta interface{}) erro
 	var domainConnAll [][]string
 	var attachedGWAll []string
 	var attachedVPCAll [][]string
+
+	mapSecurityDomainRule := make(map[string][3]bool)
+
 	domains := d.Get("security_domains").([]interface{})
 	for _, domain := range domains {
 
 		dn := domain.(map[string]interface{})
 		domainsAll = append(domainsAll, dn["security_domain_name"].(string))
 
+		var afd, ned, nfd bool
+		if dn["aviatrix_firewall"] == nil {
```

Comment:
do we need this? since its bool?

Reply:
Only for convenience, can get rid of it.